lib/xcodeproj/project/object/helpers/sort_helper.rb

```diff
@@ -0,0 +1,34 @@
+module Xcodeproj
+  # Wrapper for a string that sorts by name like Xcode does.
+  # @example
+  #   arrayOfFilenames.sort_by { |s| XcodeSortString.new(s) }
+  class XcodeSortString
+    include Comparable
+    attr_reader :str, :scrubbed, :ints_and_strings, :i_s_pattern
+
+    def initialize(str)
+      @str = str
+      @scrubbed = str.downcase
+      # `Integer`+`rescue`: credit to https://stackoverflow.com/a/39025521
+      @ints_and_strings = @scrubbed.scan(/\d+|\D+/).map { |s| begin Integer(s, 10); rescue ArgumentError; s end }
+      # comparing patterns: credit to https://rosettacode.org/wiki/Natural_sorting#Ruby
+      @i_s_pattern = @ints_and_strings.map { |el| el.is_a?(Integer) ? :i : :s }.join
+    end
+
+    def <=>(other)
+      if i_s_pattern.start_with?(other.i_s_pattern) || other.i_s_pattern.start_with?(i_s_pattern)
+        compare = ints_and_strings <=> other.ints_and_strings
+        if compare != 0
+          # we sort naturally
+          compare
+        else
+          # equality, we sort reverse raw
+          -(str <=> other.str)
```

Comment:
to sort in reverse order, you can do `other.str <=> str` with no need for negation